docs/datasource_api_userguide.md

```diff
@@ -1,147 +1,161 @@
-
-See [code examples with extensions](https://github.com/pingcap/tispark-test/blob/master/tispark-examples/src/main/scala/com/pingcap/tispark/examples/TiDataSourceExampleWithExtensions.scala).
-
-### init SparkConf
-```scala
-val sparkConf = new SparkConf()
-  .setIfMissing("spark.master", "local[*]")
-  .setIfMissing("spark.app.name", getClass.getName)
-  .setIfMissing("spark.sql.extensions", "org.apache.spark.sql.TiExtensions")
-  .setIfMissing("spark.tispark.pd.addresses", "pd0:2379")
-  .setIfMissing("spark.tispark.tidb.addr", "tidb")
-  .setIfMissing("spark.tispark.tidb.password", "")
-  .setIfMissing("spark.tispark.tidb.port", "4000")
-  .setIfMissing("spark.tispark.tidb.user", "root")
-
-
-val spark = SparkSession.builder.config(sparkConf).getOrCreate()
-val sqlContext = spark.sqlContext
-```
-
-### Read using scala
-```scala
-// use tidb config in spark config if does not provide in data source config
-val tidbOptions: Map[String, String] = Map()
-val df = sqlContext.read
-  .format("tidb")
-  .options(tidbOptions)
-  .option("database", "tpch_test")
-  .option("table", "CUSTOMER")
-  .load()
-  .filter("C_CUSTKEY = 1")
-  .select("C_NAME")
-df.show()
-```
-
-### Write using scala
-```scala
-/* create table before run the code
- CREATE TABLE tpch_test.target_table_orders (
-   `O_ORDERKEY` int(11) NOT NULL,
-   `O_CUSTKEY` int(11) NOT NULL,
-   `O_ORDERSTATUS` char(1) NOT NULL,
-   `O_TOTALPRICE` decimal(15,2) NOT NULL,
-   `O_ORDERDATE` date NOT NULL,
-   `O_ORDERPRIORITY` char(15) NOT NULL,
-   `O_CLERK` char(15) NOT NULL,
-   `O_SHIPPRIORITY` int(11) NOT NULL,
-   `O_COMMENT` varchar(79) NOT NULL
- )
-*/
-
- // use tidb config in spark config if does not provide in data source config
- val tidbOptions: Map[String, String] = Map()
-
- // data to write
- val df = sqlContext.read
-   .format("tidb")
-   .options(tidbOptions)
-   .option("database", "tpch_test")
-   .option("table", "ORDERS")
-   .load()
-
- // Append
- df.write
-   .format("tidb")
-   .options(tidbOptions)
-   .option("database", "tpch_test")
-   .option("table", "target_table_orders")
-   .mode("append")
-   .save()
-```
-
-### Use another TiDB
-TiDB config can be overwrite in data source options, thus one can connect to a different TiDB.
-
-```scala
-// tidb config priority: data source config > spark config
-val tidbOptions: Map[String, String] = Map(
-  "tidb.addr" -> "anotherTidbIP",
-  "tidb.password" -> "",
-  "tidb.port" -> "4000",
-  "tidb.user" -> "root",
-  "spark.tispark.pd.addresses" -> "pd0:2379"
-)
 
-val df = sqlContext.read
-  .format("tidb")
-  .options(tidbOptions)
-  .option("database", "tpch_test")
-  .option("table", "CUSTOMER")
-  .load()
-  .filter("C_CUSTKEY = 1")
-  .select("C_NAME")
-df.show()
-```
+Because TiDB is a database that supports `transaction`, The TiDB Connector for Spark also supports `transaction`, which means that:
```

Comment:
This pretty weird. @birdstorm We need to discuss the entire paragraph.

Reply:
That's what I think. We will discuss the entire chapter before reaching an agreement.

Reply:
i will fix in next PR.